pkg/endpointmanager/conntrack.go

```diff
@@ -137,30 +133,23 @@ func EnableConntrackGC(ipv4, ipv6 bool) {
 	}()
 }
 
-// RmCTEntriesOf cleans the connection tracking table of the given endpoint `e`
-// by removing the CT's entries that have the src_sec_id equal to any of keys
-// in the ids' map.
-func RmCTEntriesOf(ipv4Enabled bool, e *endpoint.Endpoint, ids map[uint32]bool) {
-	var eIPv4, eIPv6 net.IP
-
-	e.Mutex.RLock()
-	isLocal := e.Opts.IsEnabled(endpoint.OptionConntrackLocal)
-	eIPv6 = e.IPv6.IP()
-	if ipv4Enabled {
-		eIPv4 = e.IPv4.IP()
-	}
-	// We can unlock the endpoint mutex since
-	// in runGC it will be locked as needed.
-	e.Mutex.RUnlock()
+// RmCTEntriesOf cleans the connection tracking table of the given endpoint `e`.
+// It removes all CT entries that of the CT table local or global, defined by isLocal,
+// that contains:
+//  - all the IP addresses given in the ips slice AND
+//  - any of the given ids in the ids map, maps to true and matches the
+//    src_sec_id in the CT table.
+func RmCTEntriesOf(ipv4Enabled bool, e *endpoint.Endpoint, isLocal bool, ips []net.IP, ids policy.RuleContexts) {
```

Comment:
The next commit removes all callers of this except the ones introduced in this commit. In what scenarios is `isLocal` and `ips` not derived from the endpoint that is also passed in? Can we just remove those parameters to simplify this?

Reply:
There are no scenarios, I just want to prevent locking the endpoint on this function when I need to read the endpoints's IPs and options. The endpoint might be locked to update it's status but's only going to be executed in case of a failure.